torch_xla/experimental/xla_sharding.py

```diff
@@ -399,6 +400,14 @@ def mark_sharding(t: Union[torch.Tensor, XLAShardedTensor], mesh: Mesh,
     f"Each device mesh dimension should appear at most once in partition_spec {partition_spec}."
 
   tile_assignment = _get_tile_assignment(mesh)
+  # check for sharding 2D tensor on a 3D mesh
+  original_shape = tuple(t.shape)
+  tensor_expand = 0
+  # if len(mesh.shape()) - len(partition_spec) >= 1:
+  while tensor_expand < len(mesh.shape()) - len(partition_spec):
```

Comment:
Instead of doing multiple expand in a loop, how about we calculate the sizes first and then pass that to a single expand?